extensions/wikia/AdEngine/js/PageFairDetection.js

```diff
@@ -0,0 +1,53 @@
+define('ext.wikia.adEngine.pageFairDetection', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.utils.scriptLoader',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, scriptLoader, doc, log, window) {
+	'use strict';
+
+	var context = adContext.getContext(),
+		logGroup = 'ext.wikia.adEngine.pageFairDetection';
+
+	window.pf_notify = detector;
+	window.bm_website_code = getWebsiteCode();
+
+	function getWebsiteCode() {
+		return context.opts.pageFairWebsiteCode;
+	}
+
+	function detector(adblockDetected) {
+		var event,
+			eventName = adblockDetected ? 'pf.blocking' : 'pf.not_blocking';
+
+		if (!context.opts.pageFairDetection) {
+			log('PageFair disabled', 'debug', logGroup);
+			return null;
+		}
+
+		log(['PageFair detection, adBlock detected:', adblockDetected], 'debug', logGroup);
+
+		window.ads.runtime.pf = window.ads.runtime.pf || {};
+		window.ads.runtime.pf.blocking = adblockDetected;
+
+		try {
+			event = new Event(eventName);
```

Comment:
Actually, it is, that's why I created this try-catch construction, I noticed only the incompatibility with Phantom, but now I saw that it is not supported by some browsers.

So is construction from `catch` better?